modin/core/io/text/text_file_dispatcher.py

```diff
@@ -1057,8 +1092,21 @@ def _read(cls, filepath_or_buffer, **kwargs):
                 newline=newline,
                 header_size=header_size,
                 pre_reading=pre_reading,
+                read_callback_kw=read_callback_kw
+                if not compute_metadata_before_skipping_rows
+                else None,
             )
+            if not compute_metadata_before_skipping_rows:
+                pd_df_metadata = pd_df_metadata_temp
 
+        column_names = pd_df_metadata.columns
+        column_widths, num_splits = cls._define_metadata(pd_df_metadata, column_names)
+        partition_kwargs.update(
+            dict(
+                num_splits=num_splits,
+                names=column_names if use_inferred_column_names else names,
+            )
+        )
```

Comment:
```suggestion
        partition_kwargs["num_splits"] = num_splits
        partition_kwargs["names"] = column_names if use_inferred_column_names else names
```
I think this reads easier